### ISSUE TITLE
Different versions of YOLOv8s have large differences in map50 for training Visdrone datasets

### ISSUE THREAD

**Issue by Bigtuo** [ultralytics/ultralytics#17364]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Different versions of YOLOv8s have large differences in map50 for training Visdrone datasets. What is the reason for this?
Version v8.0.213---YOLOv8s training 100epochs---map50 is 0.410
Version v8.1.47---YOLOv8s training 100epochs---map50 is 0.36
Version v8.3.13---YOLOv8s training 100epochs---map50 is 0.332

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Bigtuo, thank you for bringing this to our attention! 🚀 This is an automated response to let you know that an Ultralytics engineer will assist you soon.

We recommend checking our [Docs](https://docs.ultralytics.com) where you might find relevant information about the different YOLO versions and their expected performance. If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us diagnose the issue effectively.

For custom training queries like yours, it is helpful to include detailed information such as dataset image examples, training logs, and confirmation that you're adhering to our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

To stay connected and get support from the Ultralytics community, join our [Discord](https://ultralytics.com/discord) 🎧 for real-time chat or engage in deeper discussions on [Discourse](https://community.ultralytics.com). You can also check out our [Subreddit](https://reddit.com/r/ultralytics) for more community-driven insights.

## Upgrade

Make sure you're running the latest `ultralytics` package along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to rule out issues that might have already been resolved:

```bash
pip install -U ultralytics
```

## Environments

You can run YOLO in one of these updated environments that have all necessary dependencies, such as [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/), preconfigured:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

By checking this badge, you can verify if all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. These tests ensure the integrity of different YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across various platforms including macOS, Windows, and Ubuntu, updating every 24 hours and with every commit.

**Comment by Bigtuo** [comment]
![results](https://github.com/user-attachments/assets/ff8b5ebc-3e45-484b-95ee-28db275fd7ff)
Version v8.0.213---YOLOv8s training 100epochs---map50 is 0.410

**Comment by Bigtuo** [comment]
![results](https://github.com/user-attachments/assets/90f819d7-8c71-4a01-a772-2a18d315e55e)
Version v8.3.13---YOLOv8s training 100epochs---map50 is 0.332

**Comment by Bigtuo** [comment]
![results](https://github.com/user-attachments/assets/3f7bb984-dea2-4092-8c27-58c1ec8a94a8)
Version v8.3.27---YOLOv8s training 100epochs---map50 is 0.346

**Comment by Bigtuo** [comment]
The content of **train.py** is as follows：

import os
os.environ['CUDA_VISIBLE_DEVICES'] = '0'
from ultralytics import YOLO

model = YOLO("yolov8s.pt")
model.train(data="VisDrone.yaml", imgsz=640, epochs=100, workers=4, batch=8, close_mosaic=10, cache=True）
metrics = model.val()

**Comment by glenn-jocher** [comment]
Thank you for sharing your script. The variations in mAP50 across different YOLOv8 versions could be due to changes in model architecture, hyperparameters, or training optimizations. Ensure you're using consistent settings and consider reviewing the release notes for each version to understand any impactful changes.

**Comment by Bigtuo** [comment]
Except for resetting the values of epoch and batch size, these experiments are all trained directly after downloading the corresponding version of the source code. The hyperparameters are all default values, and pre-trained weights are loaded at the same time. In addition, during the training process, the intermediate weight size of the old version (with map50 of 0.41) is more than 80+MB, while the weight size of the new version is only 40+MB. After the training is completed, they are both 20+MB. What is the cause of this difference?



---Original---
From: "Glenn ***@***.***&gt;
Date: Wed, Nov 6, 2024 21:23 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [ultralytics/ultralytics] Different versions of YOLOv8s havelarge differences in map50 for training Visdrone datasets (Issue #17364)




 
Thank you for sharing your script. The variations in mAP50 across different YOLOv8 versions could be due to changes in model architecture, hyperparameters, or training optimizations. Ensure you're using consistent settings and consider reviewing the release notes for each version to understand any impactful changes.

**Comment by glenn-jocher** [comment]
The difference in weight sizes between versions may be due to changes in model architecture or optimization techniques. Reviewing the release notes for each version can provide insights into these differences.

**Comment by Bigtuo** [comment]
Hello, I still can't find the reason why map50 is reduced, but I tested other findings:
1. The difference in model weights is because in versions 8.2.0 and above, def save_model(self): sets model to None, and the optimizer parameter is set to fp16. Therefore, the model size is 85MB and 40+MB during training;
2. Version 8.3.13 has lower training accuracy because YOLOv8_head uses YOLOv11_head;
3. After training version 8.1.0, it was found that this version trained Visdrone2019, and map50 could reach 0.40+, while mp50 of versions 8.2.0 and above was between 0.35-0.37.

**Comment by Bigtuo** [comment]
> The difference in weight sizes between versions may be due to changes in model architecture or optimization techniques. Reviewing the release notes for each version can provide insights into these differences.

Hello, author!
I will train and test the Visdrone2019 dataset on multiple versions later, and finally found that starting from version 8.1.45, when the training parameter cache=True, map50 is only about 0.35, but when I set cache=False, map50 can reach 0.41.

Later, locate def load_image(self, i, rect_mode=True): in ultralytics/data/base.py, and comment out the code if self.cache != "ram":. At this time, the training parameter cache=True, map50 can also reach 0.41.

Since there is no update log for this version, I want to know why I added the code if self.cache != "ram": in the first place.

Looking forward to your reply!

**Comment by glenn-jocher** [comment]
@Bigtuo thank you for your detailed observations. The addition of `if self.cache != "ram":` likely aimed to optimize data loading by differentiating caching strategies. However, it seems to have impacted mAP50 in your case. For specific changes, reviewing the code history in the repository might provide insights.

**Comment by Y-T-G** [comment]
`cache` can be either "ram" or "disk". If you use `True`, then it uses "ram" caching.

**Comment by Y-T-G** [comment]
If you remove that line, it won't cache in RAMA
https://github.com/ultralytics/ultralytics/pull/9500#issuecomment-2033453138

https://github.com/ultralytics/ultralytics/pull/9828

**Comment by nemuwww** [comment]
我在用yolo11和yolov8训练visdrone时遇到了同样的问题，yolov8s的mAP50可以达到39%，但是yolo11只有30%,而且收敛十分缓慢。
